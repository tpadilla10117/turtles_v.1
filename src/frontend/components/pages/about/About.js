import React from 'react';
import {
  Footer,
  HeroCarousel,
  InfoSection,
  GridInfoSection
} from '../../utils.js';

import { heroCarouselData } from '../../../seed';
import AmandaSmiling from '../../../assets/images/Amanda-04.jpeg';
import Amanda2 from '../../../assets/images/Amanda-46.jpeg';
import Amanda3 from '../../../assets/images/Amanda-62.jpeg';

function About() {
  return (
    <div className='aboutpg-parent-container'>
        
        <HeroCarousel imgData={heroCarouselData} />
        <InfoSection 
          ptext={'"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.'}
          ptext2={'"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.'} 
          ptext3={'"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.'}
          headingtxt={'Welcome to...'}
          infoSectionClassName={'aboutpg-infoSection-parent-container'}
          infoSectionImg={AmandaSmiling}
          alt={'Somne sample text'}
          infoSectionTxtWrapperClassName={'aboutpg-infoSection-txt-parent-container'}
        
        />
        <GridInfoSection
          gridInfoSectionClassName={'aboutpg-gridInfoSection-parent-container'}
          gridInfoSectionTxtParent={'aboutpg-gridInfoSection-txt-parent-container'}
          gridInfoSectionHeadingContainer={'aboutpg-gridInfoSection-h2-parent-container'}
          gridInfoSectionTxtWrapper1ClassName={'aboutpg-gridInfoSection-txt-wrapper1'}
          gridInfoSectionTxtWrapper2ClassName={'aboutpg-gridInfoSection-txt-wrapper2'}
          ptext={'Turtles has a lot to offer to her students whether they are novice or seasoned. Turtles has a lot to offer to her students whether they are novice or seasoned.'}
          ptext2={'With over 5 years in industry, Turtles has gained a wealth of experience in many areas over a long time. Turtles has a lot to offer to her students whether they are novice or seasoned.'}
          h1text={'Born To Perform'}
          h1text2={'Simply Outstanding'}
          gridInfoSectionImg={Amanda2}
          gridInfoSectionImg2={Amanda3}
        />
        <Footer />
    </div>
  )
}

export default About