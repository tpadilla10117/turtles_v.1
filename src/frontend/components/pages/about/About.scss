.aboutpg-parent-container {

    .aboutpg-infoSection-parent-container {
        width: 100%;
        height: 100%;
        background-color: $background-complement;
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
        padding: 50px;

        @media screen and (min-width: 760px) {
           display: flex;
           flex-direction: row;
        
        }
    
        @media screen and (min-width: 1024px) {
            height: 100%;
            
        }

        .aboutpg-infoSection-txt-parent-container {
            /* border: 1px solid green; */
            padding: 2rem;
            
        }

        .infoSection-img-parent {
            display: none;
            /* filter: drop-shadow(0px 4px 4px rgb(0 0 0 / 50%) ); */
    
            @media screen and (min-width: 760px) {
                display: block;
                width: 100%;
                height: 100%;
                
                /* position: relative; */
                
            }
    
            .infoSection-img {
                @media screen and (min-width: 760px) {
                    display: block;
                   /* border: 1px solid yellow; */
                   object-fit: cover;
                   object-position: center;
                   width: 100%;
                   position: relative;
                   height: 650px;
                   border-radius: 5px;
                   
                }

                @media screen and (min-width: 1500px) {
                    height: 850px;
                    object-position: center;
                }
            }

        }

        .infoSection-img-parent.slide-in {
            animation: fadeInRight;
            animation-duration: 1s;
            /* animation-delay: .1s; */
    
            @keyframes fadeInRight {
                from {
                opacity: 0;
                transform: translate3d(100%, 0%, 0);
                }
            
                to {
                opacity: 1;
                transform: translate3d(0, 0, 0);
                }
            }
        }

        div {
            width: 100%;
            height: 100%;
            text-align: left;
            position: relative;
            /* border: 1px solid green; */
            /* align-self: center; */
            /* margin-top: 5px; */
    
            @media screen and (min-width: 760px) {
                /* border: 1px solid green; */
                /* display: flex;
                flex-direction: column;
                justify-content: center; */
                
            }
    
            h1 {
                font-weight: 700;
                font-family: $header-primaryfont;
                font-size: $heading-primary;
                color: $text-primary;
                text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
                margin-bottom: 40px;
    
                @media screen and (min-width: 760px) {
                    margin-bottom: 20px;
                    
                }
    
                @media screen and (min-width: 1024px) {
                    font-size: 40px;
                    margin-bottom: 50px;
                    
                }
                @media screen and (min-width: 1366px) {
                    font-size: 50px;
                    
                }
                
            }
    
            .infoSection-h1.fadeInUp {
                animation: fadeInUp;
                animation-duration: 1s;
                /* animation-delay: .5s; */
            
                    @keyframes fadeInUp {
                        from {
                        opacity: 0;
                        transform: translate3d(0, 100%, 0);
                        }
                    
                        to {
                        opacity: 1;
                        transform: translate3d(0, 0, 0);
                        }
                    }
            }
            
    
            p {
                color: $text-primary;
                font-family: $paragraph-lightfont;
                font-weight: 400;
                font-size: $paragraph-primary;
                text-align: left;
    
                @media screen and (min-width: 760px) {
                    font-size: 12px;
                    
                }
    
                @media screen and (min-width: 1024px) {
                    font-size: 16px;
                    
                }
                @media screen and (min-width: 1366px) {
                    font-size: 20px;
                    
                }
            }
    
    
            /* .infoSection-p1, .infoSection-p2 {
                opacity: 0;
    
                &.active {
                    opacity: 1;
                }
            } */
    
    
            .infoSection-p1.fadeInUp {
                animation: fadeInUp;
                animation-duration: 1s;
                /* animation-delay: .5s; */
            
                    @keyframes fadeInUp {
                        from {
                        opacity: 0;
                        transform: translate3d(0, 100%, 0);
                        }
                    
                        to {
                        opacity: 1;
                        transform: translate3d(0, 0, 0);
                        }
                    }
            }
    
            .infoSection-p2.fadeInUp {
                animation: fadeInUp;
                animation-duration: 1s;
                /* animation-delay: 1s; */
            
                    @keyframes fadeInUp {
                        from {
                        opacity: 0;
                        transform: translate3d(0, 100%, 0);
                        }
                    
                        to {
                        opacity: 1;
                        transform: translate3d(0, 0, 0);
                        }
                    }
            }
            .infoSection-p3.fadeInUp {
                animation: fadeInUp;
                animation-duration: 1s;
                /* animation-delay: 1s; */
            
                    @keyframes fadeInUp {
                        from {
                        opacity: 0;
                        transform: translate3d(0, 100%, 0);
                        }
                    
                        to {
                        opacity: 1;
                        transform: translate3d(0, 0, 0);
                        }
                    }
            }
    
            .infoSection-contact-btn {
                position: absolute;
                top: 95%;
                left: 50%;
                transform: translate(-50%, -50%);
            }
        }

    }
}