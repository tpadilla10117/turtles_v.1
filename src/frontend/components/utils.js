/* This components handles exports in the main App: */

    /* "layout" components: */
        export { default as Nav } from '../components/layout/nav/Nav.jsx';

        export { default as DesktopNav } from '../components/layout/desktopNav/DesktopNav.jsx';

        export { default as Sidebar } from '../components/layout/sidebar/Sidebar.jsx';

        export { default as Header } from '../components/layout/header/Header.jsx';

        export { default as Footer } from '../components/layout/footer/Footer.jsx';



/* -------------------------------------------------------------------------- */


    /* "pages" components: */

        export { default as Landing } from '../components/pages/landing/Landing.js';

        export { default as About } from '../components/pages/about/About.js';

        export { default as Gallery } from '../components/pages/gallery/Gallery.js';

        export { default as Contact } from '../components/pages/contact/Contact.js';

        export { default as InfoSection } from '../components/layout/infoSection/InfoSection.jsx';


/* -------------------------------------------------------------------------- */
/* Functions: */

/* Scroll to top of the page after a user clicks a link: */
export const scrollTop = () => {
    window.scrollTo({ behavior: "smooth", top: "0px"});
};