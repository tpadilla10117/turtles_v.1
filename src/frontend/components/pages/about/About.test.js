import { shallow } from 'enzyme';
import React from 'react';
import About from './About';
import { HeroCarousel } from '../../utils'; 

describe('The About.js component:', () => {

    it('expect to render the About.js component', () => {
        expect.assertions(2);
        expect(shallow(<About />)).toMatchSnapshot();
        expect(shallow(<About />).length).toEqual(1);
    });


    it('expect to render the HeroCarousel', () => {
        const mockHeroCarouselData = [
            {
                title: 'xxx',
                caption: 'some text',
                alt: 'an alternative xxx',
                thumbnail: 'xxx',
                buttonId: 1,
                id: 1
            }
        ];

        expect.assertions(2);

        expect(shallow(<HeroCarousel imgData={mockHeroCarouselData} />)).toMatchSnapshot();

        expect(shallow(<HeroCarousel imgData={mockHeroCarouselData} />).length).toEqual(1);
    });
});