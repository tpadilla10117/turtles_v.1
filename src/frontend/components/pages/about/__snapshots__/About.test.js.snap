// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The About.js component: expect to render the About.js component 1`] = `ShallowWrapper {}`;

exports[`The About.js component: expect to render the HeroCarousel 1`] = `ShallowWrapper {}`;
